app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,34 @@
+class CategoriesController < ApplicationController
+  include Pagy::Backend
+  def index
+    @pagy, @categories = pagy(Category.includes(:products),
+                              items: Settings.page_size)
+  end
+
+  def show
+    @category = find_category
+    return unless @category
+
+    load_products
+  end
+
+  private
+
+  def path_args page
+    [@category, {page:}]
+  end
+
+  def find_category
+    category = Category.find_by(id: params[:id])
+    unless category
+      flash[:alert] = t("categories.not_found")
+      redirect_to categories_path
+    end
+    category
```

Comment:
```suggestion
    @category = Category.find_by(id: params[:id])
    return if  @category.present?
      flash[:alert] = t "categories.not_found"
      redirect_to categories_path
```
Em dùng thêm before action nhé